spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/KubernetesConfigDataLoader.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.commons.config;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.boot.context.config.ConfigData;
+import org.springframework.boot.context.config.ConfigData.Option;
+import org.springframework.boot.context.config.ConfigDataLoader;
+import org.springframework.boot.context.config.ConfigDataLoaderContext;
+import org.springframework.boot.context.config.ConfigDataResourceNotFoundException;
+import org.springframework.core.Ordered;
+import org.springframework.core.env.PropertySource;
+
+/**
+ * @author Ryan Baxter
+ */
+public class KubernetesConfigDataLoader implements ConfigDataLoader<KubernetesConfigDataResource>, Ordered {
+
+	@Override
+	public ConfigData load(ConfigDataLoaderContext context, KubernetesConfigDataResource resource)
+			throws IOException, ConfigDataResourceNotFoundException {
+		List<PropertySource<?>> propertySources = new ArrayList<>();
+		ConfigMapPropertySourceLocator configMapPropertySourceLocator = context.getBootstrapContext()
```

Comment:
I am slightly confused about this line for example:

```
SecretsPropertySourceLocator secretsPropertySourceLocator = context.getBootstrapContext()
				.getOrElse(SecretsPropertySourceLocator.class, null);
```

Documentation of `getOrElse` says : 

> Return an instance from the context if the type has been registered. **The instance will be created** it if it hasn't been accessed previously.

As such, I think I understand why you are calling `getOrElse(SecretsPropertySourceLocator.class, null);` (with a `null`): you do not want to register such an instance, because users might disable this via `spring.cloud.kubernetes.secrets.enabled=false`. So you want to know if such an instance is registered or not, really, but there is a method for that : `ConfigurableBootstrapContext::isRegistered`. 

May be I miss something (I am finding out about config-data api as I read this code :) ), but can we use that instead?

Reply:
Makes perfect sense :)